serde_derive/src/fragment.rs

```diff
@@ -73,3 +73,88 @@ impl ToTokens for Match {
         }
     }
 }
+
+use internals::attr::StrBoolInt as StrBoolIntInternal;
+
+/// An extended name, used for the type tag for enums.
+#[derive(Debug, Clone)]
+pub enum StrBoolInt {
+    Str(String),
+    Bool(bool),
+    Int(u64),
+}
+
+impl StrBoolInt {
+    /// Get a string description for use in error messages.
+    ///
+    /// It will be used as second argument in
+    /// `_serde::Serializer::serialize_struct(__serializer, name, len)`
+    /// while serializing the inner struct in adjacently tagged enums.
```

Comment:
This may be a problem, depending on how a serializer serializes a struct name. But I guess that formats that require a tagged enum do not serialize the variant/struct name. (This case is part of the tests and has a comment there also)